Rails workshop

praweb opened the pull request:
@categories: crud operations on categories got implemented.

@posts: create and view are in progress

Commit:
Crud operations using Rails

Implemented crud functionality to Categories
@posts: create operation is in progress. Radio button functionality is
in progress

Commit:
crud operations on categories

@missing files for categories.

krisquigley commented on app/views/categories/index.html.erb:
```diff
@@ -1 +1,12 @@
+
+<% categories.each do |category| %>
+	<div>
+		<%= link_to category.name, {action: :show, id: category} %>
```
Use URL helpers where you can i.e. `category_path(category)`

krisquigley commented on app/views/categories/show.html.erb:
```diff
@@ -0,0 +1 @@
+<%= category.name%>
```
Can you also pull in all the posts associated with the categories here?

krisquigley commented on app/views/posts/index.html.erb:
```diff
@@ -0,0 +1,29 @@
+	</h5>
+	<br/><br/>
+	<!-- Form to create the post -->
+	<%= form_for post do |f| %>
```
This form could be put into a partial for reuse.

krisquigley commented on app/views/posts/show.html.erb:
```diff
@@ -0,0 +1,13 @@
+	<div>
+		<%= post.inspect %>
+		<span class="label label-default">  Category Id
+			<%=  post.category_id %>
```
To show the category you can do `post.category.name`

Commit:
Post creation, view, and comments creation and view

@posts : creation of posts, viewing posts, deletion of posts has been
done

@comments: creation and viewing of comments are done

#pending: deletion of comments, and listing of posts basing on
categories, validations, refresh of pages on back navigation, few more
details display in each page, published validations.

krisquigley commented on app/views/posts/index.html.erb:
```diff
@@ -0,0 +1,23 @@
+	<br/><br/>
+	
+	<!-- listing the posts if available -->
+	<% if posts.length>0 %>
```
This can just be `if posts` or `if !posts.empty?`

krisquigley commented on app/views/posts/index.html.erb:
```diff
@@ -0,0 +1,23 @@
+			<div>
+				<%= link_to post.title, post_path(post), class:'col-xs-12' %>
+				<span>
+					<%= link_to 'edit', {action: :edit, id: post}, class:'btn-xs btn-primary'%>
```
Can you use URL helpers here too please

krisquigley commented on app/views/posts/show.html.erb:
```diff
@@ -0,0 +1,22 @@
+	<!-- Listing all the available comments -->
+	<%= render './comments/comments' , collection: @comments %>
+	<!-- Adding comments to the posts -->
+	<%= render './comments/new' , collection: post.comments %>
```
You should just be able to pass `<%= render 'comments/new' , collection: post.comments %>`
